src/scripting/game_lua_kernel.cpp

```diff
@@ -3183,37 +3193,42 @@ int game_lua_kernel::intf_progress_achievement(lua_State *L)
 	int amount = luaL_checkinteger(L, 3);
 	int limit = luaL_checkinteger(L, 4);
 
-	config cfg;
-	cfg["progress"] = 0;
-	cfg["max_progress"] = 0;
-
 	for(achievement_group& group : game_config_manager::get()->get_achievements()) {
 		if(group.content_for_ == content_for) {
 			for(achievement& achieve : group.achievements_) {
 				if(achieve.id_ == id) {
-					// found the achievement
+					// check that this is a progressable achievement
+					if(achieve.max_progress_ == 0) {
+						ERR_LUA << "Attempted to progress achievement " << id << " for achievement group " << content_for << ", which does not have max_progress set.";
+						lua_pushinteger(L, -2);
```

Comment:
That's not mentioned in the doc comment.

Reply:
Updated.